opentelemetry-sdk/benches/metrics_counter.rs

```diff
@@ -57,62 +57,72 @@ fn criterion_benchmark(c: &mut Criterion) {
-                    KeyValue::new("attribute2", ATTRIBUTE_VALUES[index_second_attribute]),
-                    KeyValue::new("attribute3", ATTRIBUTE_VALUES[index_third_attribute]),
-                    KeyValue::new("attribute4", ATTRIBUTE_VALUES[index_fourth_attribute]),
-                ],
-            );
-        });
+        b.iter_batched(
+            || {
+                // 4*4*10*10 = 1600 time series.
+                CURRENT_RNG.with(|rng| {
+                    let mut rng = rng.borrow_mut();
+                    [
+                        rng.gen_range(0..4),
+                        rng.gen_range(0..4),
+                        rng.gen_range(0..10),
+                        rng.gen_range(0..10),
+                    ]
+                })
+            },
+            |rands| {
+                let index_first_attribute = rands[0];
+                let index_second_attribute = rands[1];
+                let index_third_attribute = rands[2];
+                let index_fourth_attribute = rands[3];
+                counter.add(
+                    1,
+                    &[
+                        KeyValue::new("attribute1", ATTRIBUTE_VALUES[index_first_attribute]),
+                        KeyValue::new("attribute2", ATTRIBUTE_VALUES[index_second_attribute]),
+                        KeyValue::new("attribute3", ATTRIBUTE_VALUES[index_third_attribute]),
+                        KeyValue::new("attribute4", ATTRIBUTE_VALUES[index_fourth_attribute]),
+                    ],
+                );
+            },
+            BatchSize::LargeInput,
+        );
     });
 }
 
 fn counter_add_unsorted(c: &mut Criterion) {
     let counter = create_counter("Counter_Add_Unsorted");
     c.bench_function("Counter_Add_Unsorted", |b| {
-        b.iter(|| {
-            // 4*4*10*10 = 1600 time series.
-            let rands = CURRENT_RNG.with(|rng| {
-                let mut rng = rng.borrow_mut();
-                [
-                    rng.gen_range(0..4),
-                    rng.gen_range(0..4),
-                    rng.gen_range(0..10),
-                    rng.gen_range(0..10),
-                ]
-            });
-            let index_first_attribute = rands[0];
-            let index_second_attribute = rands[1];
-            let index_third_attribute = rands[2];
-            let index_fourth_attribute = rands[3];
-            counter.add(
-                1,
-                &[
-                    KeyValue::new("attribute2", ATTRIBUTE_VALUES[index_second_attribute]),
-                    KeyValue::new("attribute3", ATTRIBUTE_VALUES[index_third_attribute]),
-                    KeyValue::new("attribute1", ATTRIBUTE_VALUES[index_first_attribute]),
-                    KeyValue::new("attribute4", ATTRIBUTE_VALUES[index_fourth_attribute]),
-                ],
-            );
-        });
+        b.iter_batched(
+            || {
+                // 4*4*10*10 = 1600 time series.
+                CURRENT_RNG.with(|rng| {
```

Comment:
nit: most likely obvious, but maybe leave a comment here saying iter_batched() is used to exclude setup time (rand generations) from the meaasurements.

Reply:
IMHO, it’s obvious, so I’ll leave it as is. 😀